### PR TITLE
Fixes nil pointer dereference

### PR DESCRIPTION
**Pull request by m1kola**
Fixes the following panic:

```
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]: panic: runtime error: invalid memory address or nil pointer dereference
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x2398718]
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]: goroutine 174 [running]:
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]: github.com/Azure/ARO-RP/pkg/util/clusterdata.(*workerProfilesEnricherTask).FetchData(0xc000bb2c40, 0xc000d90d20, 0xc000d90d80)
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]:         /home/vsts/work/1/s/gopath/src/github.com/Azure/ARO-RP/pkg/util/clusterdata/worker_profiles_task.go:72 +0x478
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]: github.com/Azure/ARO-RP/pkg/util/clusterdata.(*bestEffortEnricher).enrichOne.func2(0xc000837900, 0xc000bc89a0, 0xc000d90d20, 0xc000d90d80, 0x1)
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]:         /home/vsts/work/1/s/gopath/src/github.com/Azure/ARO-RP/pkg/util/clusterdata/clusterdata.go:122 +0xf1
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]: created by github.com/Azure/ARO-RP/pkg/util/clusterdata.(*bestEffortEnricher).enrichOne
May 29 13:09:11 rp-52a6c99b-000000 docker[23476]:         /home/vsts/work/1/s/gopath/src/github.com/Azure/ARO-RP/pkg/util/clusterdata/clusterdata.go:114 +0x4e4
```

To reproduce:

* Create a cluster
* `oc -n openshift-machine-api edit machineset <machineset-name>`
* Remove/comment the `template` field
* `az aro show -g $RESOURCEGROUP -n $CLUSTER`

**Actual behaviour:** RP fails with the panic above.
**Expected behaviour:** You can see cluster document as a result of the last step without `workerProfiles` field.

### CODE REVIEW

File: pkg/util/clusterdata/worker_profiles_task.go [hunk 1]
```diff
@@ -78,7 +86,10 @@ func (ef *workerProfilesEnricherTask) FetchData(callbacks chan<- func(), errs ch
 
 		machineProviderSpec, ok := o.(*azureproviderv1beta1.AzureMachineProviderSpec)
 		if !ok {
-			ef.log.Errorf("failed to read provider spec from the machine set %q: %T", machineset.Name, o)
+			// This should never happen: codecs uses scheme that has only one registered type
+			// and if something is wrong with the provider spec - decoding should fail
+			err := fmt.Errorf("failed to read provider spec from the machine set %q: %T", machineset.Name, o)
+			ef.log.Errorf(err.Error())
 			errs <- err
```
Comment (m1kola) [inline pkg/util/clusterdata/worker_profiles_task.go:93]: This change is not related to the panic we are fixing in this PR. And it shold not be a problem as it is impossible (I think) to hit this code path (hence no tests for it).

However if we were to hit it, without the change we would send a nil error to the `errs` channel here.

File: pkg/util/clusterdata/worker_profiles_task.go [hunk 2]
```diff
@@ -69,6 +69,14 @@ func (ef *workerProfilesEnricherTask) FetchData(callbacks chan<- func(), errs ch
 
 	workerProfiles := make([]api.WorkerProfile, len(machinesets.Items), len(machinesets.Items))
 	for i, machineset := range machinesets.Items {
+		if machineset.Spec.Template.Spec.ProviderSpec.Value == nil ||
+			machineset.Spec.Template.Spec.ProviderSpec.Value.Raw == nil {
+			err := fmt.Errorf("provider spec is missing in the machine set %q", machineset.Name)
+			ef.log.Errorf(err.Error())
```
Comment (asalkeld) [inline pkg/util/clusterdata/worker_profiles_task.go:75]:
```suggestion
			ef.log.Error(err)
```

File: pkg/util/clusterdata/worker_profiles_task.go [hunk 3]
```diff
@@ -70,7 +70,10 @@ func (ef *workerProfilesEnricherTask) FetchData(callbacks chan<- func(), errs ch
 	workerProfiles := make([]api.WorkerProfile, len(machinesets.Items))
 	for i, machineset := range machinesets.Items {
 		if machineset.Spec.Template.Spec.ProviderSpec.Value == nil {
-			continue
```
Comment (m1kola) [inline pkg/util/clusterdata/worker_profiles_task.go:73]: It looks like nil pointer dereference itself got fixed as part of #764. But I think we should not continue here: it will be super confusing to see 2 machinesets when you actually have 3 (one malformed).

Comment (jim-minter) [reply pkg/util/clusterdata/worker_profiles_task.go:73]: +1, the commit in #764 was a mistake on my part.

Comment (jim-minter) [reply pkg/util/clusterdata/worker_profiles_task.go:73]: Hmm, thinking about it, would it be best in this case to emit a WorkerProfile with machineProviderSpec-related fields empty?

Comment (m1kola) [reply pkg/util/clusterdata/worker_profiles_task.go:73]: `WorkerProfile` consists mostly from `machineProviderSpec` fields. We will only show name and count of VMs. But if machineset is malformed - it might mean that the count is wrong. Also confusing. I think it's better to not show `WorkerProfile`s here at all.

Comment (mjudeikis) [reply pkg/util/clusterdata/worker_profiles_task.go:73]: I think so too. Less confusion

Comment (m1kola) [reply pkg/util/clusterdata/worker_profiles_task.go:73]: Discussed in Slack. Posting for visibility:

* We want to show the desired state. Even if the desired state is unachievable (such as machine set has number of replicas specified, but no provider spec)
* Showing something to the customer is better than hiding all the machine sets when the problem is only with one

File: pkg/util/clusterdata/worker_profiles_task.go [hunk 4]
```diff
@@ -69,39 +69,41 @@ func (ef *workerProfilesEnricherTask) FetchData(callbacks chan<- func(), errs ch
...
+			workerCount = int(*machineset.Spec.Replicas)
+		}
+
+		workerProfiles[i] = api.WorkerProfile{
+			Name:  machineset.Name,
+			Count: workerCount,
+		}
+
 		if machineset.Spec.Template.Spec.ProviderSpec.Value == nil {
+			ef.log.Infof("provider spec is missing in the machine set %q", machineset.Name)
```
Comment (m1kola) [inline pkg/util/clusterdata/worker_profiles_task.go:83]: Now that we ignore the error, I changed it to info. Or should we use warning level here?

File: pkg/util/clusterdata/worker_profiles_task.go (hunk 3 above)
Comment (m1kola) [reply pkg/util/clusterdata/worker_profiles_task.go:73]: @jim-minter I updated the function and relevant test. Please take another look.

File: pkg/util/clusterdata/worker_profiles_task.go (hunk 4 above)
Comment (mjudeikis) [reply pkg/util/clusterdata/worker_profiles_task.go:83]: I think Priority6 (info/print) is good. Currently, we operate only with 3 and 6. The warning is 4. This might confuse the person looking into this.
